src/main.c

```diff
@@ -1,6 +1,182 @@
+          for(i=0; i<10; i++) {
+            sensor_values[i]=0.0;
+          }
+          strcpy(previous_timestamp, array[0]);
+          strcpy(current_timestamp, array[0]);
+          current_sensor_val = atof(array[2]);
+          sensor_values[j] = current_sensor_val;
+          LOOP_RESET = 1;
+          continue;
+        } else {
+          j++;
+          strcpy(previous_timestamp, current_timestamp);
+          strcpy(current_timestamp, array[0]);
+          current_sensor_val = atof(array[2]);
+          if(strcmp(previous_timestamp, current_timestamp)==0) {
+            sensor_values[j] = current_sensor_val;
+            continue;
+          } else {
+            // Perform sensor fusion
+            struct support_degree_matrix *spd = compute_support_degree_matrix(sensor_values);
+            struct eigen_value_vector *eigen = compute_eigen(spd);
+            double *contrib_rate = compute_contribution_rate(eigen, spd->sensor_count);
+            int contrib_rate_count = compute_contrib_rate_count(contrib_rate, 0.5, spd->sensor_count);
+            double **principal_components_matrix = compute_principal_component(spd, eigen->eigen_vector, contrib_rate_count);
+            double *integrate_support_matrix = compute_integrated_support_degree(principal_components_matrix,
+                                                   contrib_rate, contrib_rate_count, spd->sensor_count);
+            int res = eliminate_incorrect_data(integrate_support_matrix, 0.7, spd->sensor_count);
+            double *weight_coff = compute_weight_coefficient(integrate_support_matrix, spd->sensor_count);
+            double fused_value = compute_fused_output(weight_coff, sensor_values, spd->sensor_count);
+            printf("DEBUG: Time - %s, Fused Value: %f\n\n", previous_timestamp, fused_value);
+            fprintf(fptr_csv_out, "Time: %s, Fused Value: %f\n", previous_timestamp, fused_value);
+            fseek(fp,cur_pos-ct_str_length,SEEK_SET);
+            LOOP_RESET = 0;
+          }
+        }
+    }
+    
+    // Perform sensor fusion
+    struct support_degree_matrix *spd = compute_support_degree_matrix(sensor_values);
+    struct eigen_value_vector *eigen = compute_eigen(spd);
+    double *contrib_rate = compute_contribution_rate(eigen, spd->sensor_count);
+    int contrib_rate_count = compute_contrib_rate_count(contrib_rate, 0.5, spd->sensor_count);
+    double **principal_components_matrix = compute_principal_component(spd, eigen->eigen_vector, contrib_rate_count);
+    double *integrate_support_matrix = compute_integrated_support_degree(principal_components_matrix,
+                                           contrib_rate, contrib_rate_count, spd->sensor_count);
+    int res = eliminate_incorrect_data(integrate_support_matrix, 0.7, spd->sensor_count);
+    double *weight_coff = compute_weight_coefficient(integrate_support_matrix, spd->sensor_count);
+    double fused_value = compute_fused_output(weight_coff, sensor_values, spd->sensor_count);
```

Comment:
Same comment as above

Reply:
We have retained the same coding style for main file as it was by design, and the original idea of having to call the functions from main function would help in the troubleshooting and debugging since two members were operating on the same branch during the development process. Will consider this suggested style if there are no dependencies associated with the data flow

Thank you